plugins/performance-lab/includes/site-health/bfcache-compatibility-headers/helper.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Helper functions used for Cache-Control headers for bfcache compatibility site health check.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+// @codeCoverageIgnoreStart
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+// @codeCoverageIgnoreEnd
+
+/**
+ * Tests the Cache-Control headers for bfcache compatibility.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function perflab_bfcache_check_compatibility(): array {
+	$result = array(
+		'label'       => __( 'The Cache-Control response header for pages ensures fast back/forward navigation.', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => '<p>' . esc_html__( 'If the Cache-Control response header includes directives like no-store, no-cache, or max-age=0 then it can prevent instant back/forward navigations (using the browser bfcache). Your site is configured properly.', 'performance-lab' ) . '</p>',
+		'actions'     => '',
+		'test'        => 'perflab_cch_cache_control_header_check',
+	);
+
+	$response = wp_remote_get(
+		home_url( '/' ),
+		array(
+			'headers'   => array( 'Accept' => 'text/html' ),
+			'sslverify' => false,
+		)
+	);
+
+	if ( is_wp_error( $response ) ) {
+		$result['label']       = __( 'Error checking cache settings', 'performance-lab' );
+		$result['status']      = 'recommended';
+		$result['description'] = '<p>' . wp_kses(
+			sprintf(
+				/* translators: %s is the error code */
+				__( 'There was an error while checking your Cache-Control response header. Error code: <code>%s</code> and the following error message:', 'performance-lab' ),
```

Comment:
This message sounds strange. The last bit is like a sentence, but is not actually a sentence.

Reply:
```suggestion
				__( 'The request to check the Cache-Control response header responded with error code <code>%s</code> and the following error message:', 'performance-lab' ),
```